2.4/run-mongod.sh

```diff
@@ -59,36 +65,30 @@ function create_mongodb_users() {
 	# Check if the MongoDB daemon is up.
 	up_test
 
-	# Make sure env variables don't propagate to mongod process.
-	mongo_user="$MONGODB_USERNAME" ; unset MONGODB_USERNAME
-	mongo_pass="$MONGODB_PASSWORD" ; unset MONGODB_PASSWORD
-	mongo_db="$MONGODB_DATABASE" ; unset MONGODB_DATABASE
-
 	# Create MongoDB database admin, if his password is specified with MONGODB_ADMIN_PASSWORD
-	if [ "$MONGODB_ADMIN_PASSWORD" ]; then
+	if [ -v MONGODB_ADMIN_PASSWORD ]; then
 		echo "=> Creating an admin user with a ${MONGODB_ADMIN_PASSWORD} password in MongoDB"
 		mongo admin --eval "db.addUser({user: 'admin', pwd: '$MONGODB_ADMIN_PASSWORD', roles: [ 'dbAdminAnyDatabase', 'userAdminAnyDatabase' , 'readWriteAnyDatabase' ]});"
-		unset MONGODB_ADMIN_PASSWORD
 	fi
 
 	# Create standard user with read/write permissions, in specified database ('production' by default).
-	mongo $mongo_db --eval "db.addUser({user: '${mongo_user}', pwd: '${mongo_pass}', roles: [ 'readWrite' ]});"
+	mongo $MONGODB_DATABASE --eval "db.addUser({user: '${MONGODB_USERNAME}', pwd: '${MONGODB_PASSWORD}', roles: [ 'readWrite' ]});"
 	mongod -f $MONGODB_CONFIG_PATH --shutdown
 
 	# Create a empty file which indicates that the database users were created.
-	touch /var/lib/mongodb/.mongodb_users_created
+	touch /var/lib/mongodb/data/.mongodb_users_created
 
 	# Check if the MongoDB daemon is down.
 	down_test
 }
 
-test -z "$MONGODB_USERNAME" && usage
-test -z "$MONGODB_PASSWORD" && usage
-test -z "$MONGODB_DATABASE" && usage
-
-if [ ! -f /var/lib/mongodb/.mongodb_users_created ]; then
+if [ "$1" = "mongod" -a ! -f /var/lib/mongodb/data/.mongodb_users_created ]; then
```

Comment:
@mnagy imagine if user would override the the `--dbpath`, in that case the running the container would fail. Like I understand your concern but it would introduce introduce more complexity which we dont need atm. So I would rather not go this path.